### PR TITLE
Throw an error if a bootnode is registered with two different peer ids

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: core/network/src/error.rs [hunk 1]
```diff
@@ -28,13 +28,16 @@ pub enum Error {
 	Io(std::io::Error),
 	/// Client error
 	Client(client::error::Error),
+	/// Some other error occurred.
+	Other(String),
```
Comment (tomaka) [inline core/network/src/error.rs]: Why not `DuplicateBootnodeAddress` instead of `Other`?

Comment (bkchr) [reply core/network/src/error.rs]: Renamed

File: core/network/src/error.rs [hunk 2]
```diff
@@ -28,13 +28,16 @@ pub enum Error {
 	Io(std::io::Error),
 	/// Client error
 	Client(client::error::Error),
+	/// The same bootnode (based on address) is given with two different peer ids.
+	DuplicateBootnode(String),
```
Comment (tomaka) [inline core/network/src/error.rs]:
```suggestion
	DuplicateBootnode {
		address: Multiaddr,
		first_id: PeerId,
		second_id: PeerId,
	}
```

Comment (bkchr) [reply core/network/src/error.rs]: Done

**Pull request merged**